sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -847,6 +847,8 @@ private[hive] trait HiveInspectors {
 
         ObjectInspectorFactory.getStandardConstantMapObjectInspector(keyOI, valueOI, jmap)
       }
+    case Literal(_, dt: StructType) =>
+      toInspector(dt)
```

Comment:
@cxzl25, the error messages say the literal of struct type is not supported. Can you show and double check if that's supported in Hive as well?

Reply:
I use hive to run the test case sql is ok

![image](https://user-images.githubusercontent.com/3898450/59575129-9cde3280-90ec-11e9-8fcc-8229b6512e97.png)

```sql
CREATE TEMPORARY FUNCTION testLiteralStructType AS 'org.apache.hadoop.hive.ql.udf.generic.GenericUDFArray';
SELECT testLiteralStructType(named_struct('name', 'xx', 'value', 1))[0].value;
```
https://github.com/apache/spark/blob/5ae1a6bf0dea9e74ced85686ef33a87cfa3e90c2/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala#L851

Reply:
I think technically `named_struct` isn't a literal in this case. It's foldable expression in the example above. Can you fix the PR title and description accordingly to prevent confusion?

Reply:
Using ```named_struct``` just wants to construct a constant of type struct.
If use the following sql, there will be no problem mentioned in this pr.
Should I change to such a title?  Hive UDF supports foldable expression
Any suggestions?

```sql
CREATE TABLE test_struct_tbl AS 
SELECT named_struct('name', 'xx', 'value', 1) as c1;
CREATE TEMPORARY FUNCTION testLiteralStructType AS 'org.apache.hadoop.hive.ql.udf.generic.GenericUDFArray';
SELECT testLiteralStructType(c1)[0].value
FROM test_struct_tbl;
```
![image](https://user-images.githubusercontent.com/3898450/59599460-2adc0c80-9131-11e9-86ee-29d07a4c43c8.png)

![image](https://user-images.githubusercontent.com/3898450/59599235-a093a880-9130-11e9-8dc8-9d96f1eb1120.png)

Reply:
I think you can just replace "literal struct type" to "struct type foldable expression"